src/csharp/Microsoft.Spark/Sql/UDFRegistration.cs

```diff
@@ -254,7 +254,7 @@ private void Register<TResult>(string name, Delegate func)
                     name,
                     pythonFunction,
                     GetDataType<TResult>(),
-                    (int)UdfUtils.GetPythonEvalType(),
+                    (int)UdfUtils.PythonEvalType.SQL_BATCHED_UDF,
```

Comment:
I think we should allow registering VectorUdf through this API as well (this API is registering UDFs so that they can be invoked from SQL statements).

I will check PySpark and update this thread.

Reply:
I looked through the PySpark docs - https://spark.apache.org/docs/latest/api/python/pyspark.sql.html?highlight=udf#pyspark.sql.UDFRegistration.

Obviously we can't do the Python decorator pattern in C#, but it's pretty straight-forward to add a `_spark.Udf().RegisterVector` method. I didn't add an "evalType" publicly yet, since we only support one "vector" UDF evalType so far. But when we add grouped support, we can add the evalType parameter as well.